go/tasks/plugins/k8s/spark/config.go

```diff
@@ -0,0 +1,50 @@
+package spark
+
+import (
+	pluginsConfig "github.com/lyft/flyteplugins/go/tasks/config"
+	"github.com/lyft/flyteplugins/go/tasks/logs"
+)
+
+//go:generate pflags Config --default-var=defaultConfig
+
+var (
+	defaultConfig = &Config{
+		LogConfig: LogConfig{
+			UserLogs: logs.LogConfig{
+				IsKubernetesEnabled:   true,
+				KubernetesTemplateURI: "http://localhost:30084/#!/log/{{ .namespace }}/{{ .podName }}/pod?namespace={{ .namespace }}",
+			},
+		},
+	}
+
+	sparkConfigSection = pluginsConfig.MustRegisterSubSection("spark", defaultConfig)
+)
+
+// Spark-specific configs
+type Config struct {
+	DefaultSparkConfig    map[string]string `json:"spark-config-default" pflag:"-,Key value pairs of default spark configuration that should be applied to every SparkJob"`
+	SparkHistoryServerURL string            `json:"spark-history-server-url" pflag:",URL for SparkHistory Server that each job will publish the execution history to."`
+	Features              []Feature         `json:"features" pflag:"-,List of optional features supported."`
+	LogConfig             LogConfig         `json:"logs" pflag:",Config for log links for spark applications."`
```

Comment:
`*`? optional.

Reply:
I'm not sure I like the `*` to indicate optional fields in logs... the added, mostly redundant, nil checks alone are just tiring... I think we might want to add something on the tags to indicate requirement...